packages/react-vapor/src/components/validation/hoc/WithDirtyStickyFooter.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import * as _ from 'underscore';
+import {IReactVaporState} from '../../../ReactVapor';
+import {IStickyFooterProps} from '../../stickyFooter/StickyFooter';
+import {ValidationSelectors} from '../ValidationSelectors';
+import {InferableComponentEnhancer} from './connectHOC';
+
+export interface IWithDirtyStickyFooterOwnProps {
+    validationIds: string[];
+    isOpened?: boolean;
+}
+
+const mapStateToProps = (state: IReactVaporState, {validationIds}: IWithDirtyStickyFooterOwnProps) => ({
+    isDirty: ValidationSelectors.isDirty(validationIds)(state),
+});
+
+export const withDirtyStickyFooterHOC = <T extends IStickyFooterProps>(
+    Component: React.ComponentClass<T> | React.FunctionComponent<T>
+) => {
+    type NewOwnProps = Omit<T, 'isOpened'> & IWithDirtyStickyFooterOwnProps;
+    type StateProps = ReturnType<typeof mapStateToProps>;
+    const WrappedStickyFooter: React.FunctionComponent<NewOwnProps & StateProps> = ({
+        isDirty,
+        validationIds,
+        isOpened = false,
+        ...props
+    }) => {
+        return <Component {...(props as any)} isOpened={isDirty || isOpened} />;
```

Comment:
Dunno if this works?
```suggestion
        return <Component {...(props as T)} isOpened={isDirty || isOpened} />;
```

Reply:
I _did_ put `T` before. In fact, it should be `Omit<T, 'isOpened'>` since we are *removing* the obligatory `isOpened` here.

But it doesn't properly infer the type and says that it does not match the component definition...

This is me just giving up on the typing here 😢